js/views/conversation_view.js

```diff
@@ -358,7 +358,11 @@
 
             if (!this.isHidden() && !window.isFocused()) {
                 // The conversation is visible, but window is not focused
-                this.updateLastSeenIndicator({scroll: false});
+                if (!this.lastSeenIndicator) {
+                    this.resetLastSeenIndicator({scroll: false});
+                } else if (this.view.atBottom() && this.model.get('unreadCount') === this.lastSeenIndicator.getCount()) {
```

Comment:
I'll add a comment with this, but here's the reasoning: in our goal to make the position of the last seen indicator very stable, it is sometimes out of sync with the number of actually-read messages.

When it's not in sync with the the unread count, we don't want to scroll to it. But if it is in sync, and we're at the bottom of the conversation, this code will disconnect us from the bottom of the conversation if too many new messages come in, so the last seen indicator is at the top of the window and new messages appear off-screen.